File: leetcode/ds/905. Sort Array By Parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& nums) {
        vector <int> ans;
        for(auto n : nums) {
            if(n % 2 == 0) ans.push_back(n);
        }
        for(auto n : nums) {
            if(n % 2) ans.push_back(n);
        }
        return ans;
    }
};

File: leetcode/ds/905. Sort Array By Parity/905. Sort Array By Parity.rs
impl Solution {
    pub fn sort_array_by_parity(nums: Vec<i32>) -> Vec<i32> {
        let mut ans: Vec<i32> = Vec::new();
        for n in &nums {
            if n % 2 == 0 {
                ans.push(*n);
            }
        }
        for n in &nums {
            if n % 2 > 0 {
                ans.push(*n);
            }
        }
        return ans;
    }
}